base/db/R/dbfiles.R

```diff
@@ -636,3 +636,63 @@ dbfile.id <- function(type, file, con, hostname=PEcAn.remote::fqdn()) {
     invisible(NA)
   }
 }
+
+#' Function to Insert Records into the Formats and Variables Tables
+#' @name dbfile.format.variable.insert
+#' @param format_name Format Name
+#' @param header TRUE/FALSE Boolean describing whether or not the file contains a header
+#' @param skip The number of lines in the header
+#' @param mimetype_id The mimetype id of the mimetype that is associated with the format
+#' @param format_notes Any useful notes that describe the format being created
+#' @param con Connection object to connect to bety
+#' @param format_variables A TRUE/FALSE Boolean used to toggle a format-variables db query
+#' @param description variable descripthon
+#' @param units variable units
+#' @param var_notes notes to further describe the variable
+#' @param var_name variable name
+#' @param max maximum possible value
+#' @param min minimum possible value
+#' @param standard_name PEcAn specific name? 
+#' @param label ?
+#' @param type ?
+#' @author Liam Burke
+#'
+#' @return formatid
+#' @export
+#' 
+#' @examples
+#' \dontrun{
+#' bety <- betyConnect()
+#'   dbfile.format.variable.insert('Test_Format', header = TRUE, skip = 2, mimetype_id = 1060, con = bety$con, format_variables = FALSE)
+#' }
+dbfile.format.variable.insert <- function(format_name, header, skip, mimetype_id, format_notes = "", con, format_variables = FALSE, 
```

Comment:
Please drop `dbfile` from the name as this function isn't inserting anything into that table. Given that this function meant to parrallel `query.format.vars`, why not call it `insert.format.vars`?

Also, when a value is missing, such as `format_variables` we typically set it to NULL not to FALSE. For the remaining values (description, etc), pass these as a data frame to format_variables rather than passing them individually. Passing them individually is a pain and will be error prone (misalignment of vectors)

Reply:
An additional note on argument order: I recommend putting the `con` argument before `format_notes`, so that all the arguments *without* default values  come at the beginning of the call and all the arguments *with* default values come after them. 

This is partly because it's kind to the user to put "important" (~= "have to set them every time" ~= "function can't guess it") arguments first, but also so that unnamed values get matched to the required arguments before the optional ones. As currently written, the function requires a minimum of five arguments but `dbfile.format.variable.insert('Test_Format', TRUE, 2, 1060, bety$con)`  will set `format_notes` to `bety$con` and then complain that `argument "con" is missing, with no default`. 

(This second point is also a reason to name all arguments when calling any function as a user, but that ship never sailed in the R world)

Reply:
@mdietze I included `dbfile` in the name since it appeared to be a convention for all the query functions in the `dbfiles.R` document. I can change the name.

Reply:
@infotroph Thanks. Hahaha. Good point.